### PR TITLE
Panel < 0.11.0 requires bokeh < 2.3

### PR DESCRIPTION
**Pull request by AnjuSThomas-anaconda**
Added patch to Panel < 0.11.0 requires bokeh < 2.3 in main.py

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -892,6 +892,10 @@ def patch_record_in_place(fn, record, subdir):
         replace_dep(depends, 'python >=3.6', 'python >=3.7')
         replace_dep(depends, 'imagecodecs', 'imagecodecs >=2021.3.31')
 
+    # Panel<0.11.0 requires Bokeh<2.3
+    if name == 'panel':
+        if version in ['0.4.0','0.5.1','0.7.0','0.8.0'] or version.startswith('0.6.') or version.startswith('0.9.') or version.startswith('0.10.'):
```
Comment (chenghlee) [inline main.py:897]: I would not handle version this way; I would call `version.split(...)`, then use `int()` to cast and properly compare the parts. Searching this file for `.split` should give you examples of how to do that.

**Review by chenghlee — CHANGES_REQUESTED** [review]

File: main.py (hunk 1 above)
Comment (AnjuSThomas-anaconda) [reply main.py]: Thank you.. I have changed the logic

Comment (AnjuSThomas-anaconda) [reply main.py]: Replaced the logic to loop through the depends to check for all bokeh versions

Comment (AnjuSThomas-anaconda) [reply main.py]: Log of test as follows

(base) anthomas@moxies-MBP repodata-hotfixes % python test-hotfix.py main --subdir linux-64 osx-64 win-64 win-32 linux-ppc64le linux-32 noarch
downloading repodata from https://repo.anaconda.com/pkgs/main/linux-64/repodata.json
downloading repodata from https://repo.anaconda.com/pkgs/main/linux-64/repodata_from_packages.json
downloading repodata from https://repo.anaconda.com/pkgs/main/osx-64/repodata.json
downloading repodata from https://repo.anaconda.com/pkgs/main/osx-64/repodata_from_packages.json
downloading repodata from https://repo.anaconda.com/pkgs/main/win-64/repodata.json
downloading repodata from https://repo.anaconda.com/pkgs/main/win-64/repodata_from_packages.json
downloading repodata from https://repo.anaconda.com/pkgs/main/win-32/repodata.json
downloading repodata from https://repo.anaconda.com/pkgs/main/win-32/repodata_from_packages.json
downloading repodata from https://repo.anaconda.com/pkgs/main/linux-ppc64le/repodata.json
downloading repodata from https://repo.anaconda.com/pkgs/main/linux-ppc64le/repodata_from_packages.json
downloading repodata from https://repo.anaconda.com/pkgs/main/linux-32/repodata.json
downloading repodata from https://repo.anaconda.com/pkgs/main/linux-32/repodata_from_packages.json
downloading repodata from https://repo.anaconda.com/pkgs/main/noarch/repodata.json
downloading repodata from https://repo.anaconda.com/pkgs/main/noarch/repodata_from_packages.json
44415c44415
<         "bokeh >=2.2",
---
>         "bokeh >=2.2,<2.3",
44438c44438
<         "bokeh >=2.2",
---
>         "bokeh >=2.2,<2.3",
44461c44461
<         "bokeh >=2.2",
---
>         "bokeh >=2.2,<2.3",
44530c44530
<         "bokeh >=1.0.0",
---
>         "bokeh >=1.0.0,<2.3",
44550c44550
<         "bokeh >=1.1.0",
---
>         "bokeh >=1.1.0,<2.3",
44570c44570
<         "bokeh >=1.1.0",
---
>         "bokeh >=1.1.0,<2.3",
44591c44591
<         "bokeh >=1.3.0",
---
>         "bokeh >=1.3.0,<2.3",
44611c44611
<         "bokeh >=1.3.0",
---
>         "bokeh >=1.3.0,<2.3",
44631c44631
<         "bokeh >=1.3.0",
---
>         "bokeh >=1.3.0,<2.3",
44651c44651
<         "bokeh >=1.4.0",
---
>         "bokeh >=1.4.0,<2.3",
44671c44671
<         "bokeh >=1.4.0,<2.0",
---
>         "bokeh >=1.4.0,<2.3",
44692c44692
<         "bokeh >=2.0",
---
>         "bokeh >=2.0,<2.3",
44714c44714
<         "bokeh >=2.0",
---
>         "bokeh >=2.0,<2.3",
44736c44736
<         "bokeh >=2.1",
---
>         "bokeh >=2.1,<2.3",
44758c44758
<         "bokeh >=2.1",
---
>         "bokeh >=2.1,<2.3",
108150c108150
<         "bokeh >=2.2",
---
>         "bokeh >=2.2,<2.3",
108173c108173
<         "bokeh >=2.2",
---
>         "bokeh >=2.2,<2.3",
108196c108196
<         "bokeh >=2.2",
---
>         "bokeh >=2.2,<2.3",
108265c108265
<         "bokeh >=1.0.0",
---
>         "bokeh >=1.0.0,<2.3",
108285c108285
<         "bokeh >=1.1.0",
---
>         "bokeh >=1.1.0,<2.3",
108305c108305
<         "bokeh >=1.1.0",
---
>         "bokeh >=1.1.0,<2.3",
108326c108326
<         "bokeh >=1.3.0",
---
>         "bokeh >=1.3.0,<2.3",
108346c108346
<         "bokeh >=1.3.0",
---
>         "bokeh >=1.3.0,<2.3",
108366c108366
<         "bokeh >=1.3.0",
---
>         "bokeh >=1.3.0,<2.3",
108386c108386
<         "bokeh >=1.4.0",
---
>         "bokeh >=1.4.0,<2.3",
108406c108406
<         "bokeh >=1.4.0,<2.0",
---
>         "bokeh >=1.4.0,<2.3",
108427c108427
<         "bokeh >=2.0",
---
>         "bokeh >=2.0,<2.3",
108449c108449
<         "bokeh >=2.0",
---
>         "bokeh >=2.0,<2.3",
108471c108471
<         "bokeh >=2.1",
---
>         "bokeh >=2.1,<2.3",
108493c108493
<         "bokeh >=2.1",
---
>         "bokeh >=2.1,<2.3",

File: main.py [hunk 2]
```diff
@@ -892,6 +892,13 @@ def patch_record_in_place(fn, record, subdir):
         replace_dep(depends, 'python >=3.6', 'python >=3.7')
         replace_dep(depends, 'imagecodecs', 'imagecodecs >=2021.3.31')
 
+    # Panel<0.11.0 requires Bokeh<2.3
+    if name == 'panel':
+        ver_parts = version.split('.')
+        if int(ver_parts[0]) == 0 and int(ver_parts[1]) < 11:
+            for i, dep in enumerate(depends):
+                if dep.startswith('bokeh >='):
```
Comment (chenghlee) [inline main.py:900]: Look at the diff that was generated, I would make this condition `startwith('boken >=2.')` and add a separate `if` for bokeh 1.x requirements that sets the upper bound to `,<2.0.0a0`. It looks like Bokeh 2.0.0 had some breaking API changes: https://docs.bokeh.org/en/latest/docs/releases.html#release-2-0-0

**Review by chenghlee — CHANGES_REQUESTED** [review]

File: main.py (hunk 2 above)
Comment (AnjuSThomas-anaconda) [reply main.py]: Please find the test log

44687c44687
<         "bokeh >=2.2",
---
>         "bokeh >=2.2,<2.3",
44710c44710
<         "bokeh >=2.2",
---
>         "bokeh >=2.2,<2.3",
44733c44733
<         "bokeh >=2.2",
---
>         "bokeh >=2.2,<2.3",
44802c44802
<         "bokeh >=1.0.0",
---
>         "bokeh >=1.0.0,<2.0.0a0",
44822c44822
<         "bokeh >=1.1.0",
---
>         "bokeh >=1.1.0,<2.0.0a0",
44842c44842
<         "bokeh >=1.1.0",
---
>         "bokeh >=1.1.0,<2.0.0a0",
44863c44863
<         "bokeh >=1.3.0",
---
>         "bokeh >=1.3.0,<2.0.0a0",
44883c44883
<         "bokeh >=1.3.0",
---
>         "bokeh >=1.3.0,<2.0.0a0",
44903c44903
<         "bokeh >=1.3.0",
---
>         "bokeh >=1.3.0,<2.0.0a0",
44923c44923
<         "bokeh >=1.4.0",
---
>         "bokeh >=1.4.0,<2.0.0a0",
44943c44943
<         "bokeh >=1.4.0,<2.0",
---
>         "bokeh >=1.4.0,<2.0.0a0",
44964c44964
<         "bokeh >=2.0",
---
>         "bokeh >=2.0,<2.3",
44986c44986
<         "bokeh >=2.0",
---
>         "bokeh >=2.0,<2.3",
45008c45008
<         "bokeh >=2.1",
---
>         "bokeh >=2.1,<2.3",
45030c45030
<         "bokeh >=2.1",
---
>         "bokeh >=2.1,<2.3",
108745c108745
<         "bokeh >=2.2",
---
>         "bokeh >=2.2,<2.3",
108768c108768
<         "bokeh >=2.2",
---
>         "bokeh >=2.2,<2.3",
108791c108791
<         "bokeh >=2.2",
---
>         "bokeh >=2.2,<2.3",
108860c108860
<         "bokeh >=1.0.0",
---
>         "bokeh >=1.0.0,<2.0.0a0",
108880c108880
<         "bokeh >=1.1.0",
---
>         "bokeh >=1.1.0,<2.0.0a0",
108900c108900
<         "bokeh >=1.1.0",
---
>         "bokeh >=1.1.0,<2.0.0a0",
108921c108921
<         "bokeh >=1.3.0",
---
>         "bokeh >=1.3.0,<2.0.0a0",
108941c108941
<         "bokeh >=1.3.0",
---
>         "bokeh >=1.3.0,<2.0.0a0",
108961c108961
<         "bokeh >=1.3.0",
---
>         "bokeh >=1.3.0,<2.0.0a0",
108981c108981
<         "bokeh >=1.4.0",
---
>         "bokeh >=1.4.0,<2.0.0a0",
109001c109001
<         "bokeh >=1.4.0,<2.0",
---
>         "bokeh >=1.4.0,<2.0.0a0",
109022c109022
<         "bokeh >=2.0",
---
>         "bokeh >=2.0,<2.3",
109044c109044
<         "bokeh >=2.0",
---
>         "bokeh >=2.0,<2.3",
109066c109066
<         "bokeh >=2.1",
---
>         "bokeh >=2.1,<2.3",
109088c109088
<         "bokeh >=2.1",
---
>         "bokeh >=2.1,<2.3",

**Review by chenghlee — APPROVED** [review]

**Pull request merged**